emacs/.config/emacs/config.org
#+TITLE: Noah Sapse Literate Org
#+AUTHOR: Noah Sapse
#+DESCRIPTION: A literate emacs config
#+STARTUP: show2levels
#+OPTIONS: toc:3

* TABLE OF CONTENTS :toc:
- [[#load-first][LOAD FIRST]]
  - [[#follow-symlinks-automatically][Follow Symlinks Automatically]]
  - [[#elpaca-package-manager][Elpaca Package Manager]]
  - [[#evil-mode][Evil Mode]]
  - [[#keybindings-generalel][Keybindings (General.el)]]
  - [[#startup-settings][Startup Settings]]
- [[#git][Git]]
  - [[#magit][Magit]]
- [[#ivy][Ivy]]
- [[#language-support-lsp-et-al][Language Support (LSP et. al)]]
  - [[#company][Company]]
  - [[#lsp][LSP]]
- [[#org-mode][Org Mode]]
  - [[#org-bullets][Org Bullets]]
  - [[#org-capture-templates][Org Capture Templates]]
  - [[#org-modern][Org-Modern]]
  - [[#org-roam][Org Roam]]
  - [[#org-table-of-contents][Org Table of Contents]]
  - [[#org-tempo][Org Tempo]]
- [[#projectile][Projectile]]
- [[#shells-and-terminals][Shells and Terminals]]
  - [[#eshell][Eshell]]
  - [[#vterm][Vterm]]
- [[#sudo-edit][Sudo Edit]]
- [[#ui-tweaks][UI Tweaks]]
  - [[#fonts][Fonts]]
  - [[#icons][Icons]]
  - [[#line-numbers][Line Numbers]]
  - [[#modeline-doom-modeline][Modeline (Doom Modeline)]]
  - [[#themes][Themes]]
- [[#which-key][Which Key]]

* LOAD FIRST

** Follow Symlinks Automatically
#+begin_src emacs-lisp
  (setq vc-follow-symlinks t)
#+end_src

** Elpaca Package Manager

#+BEGIN_SRC emacs-lisp

  (defvar elpaca-installer-version 0.7)
  (defvar elpaca-directory (expand-file-name "elpaca/" user-emacs-directory))
  (defvar elpaca-builds-directory (expand-file-name "builds/" elpaca-directory))
  (defvar elpaca-repos-directory (expand-file-name "repos/" elpaca-directory))
  (defvar elpaca-order '(elpaca :repo "https://github.com/progfolio/elpaca.git"
			  :ref nil
			  :files (:defaults (:exclude "extensions"))
			  :build (:not elpaca--activate-package)))
  (let* ((repo  (expand-file-name "elpaca/" elpaca-repos-directory))
   (build (expand-file-name "elpaca/" elpaca-builds-directory))
   (order (cdr elpaca-order))
   (default-directory repo))
    (add-to-list 'load-path (if (file-exists-p build) build repo))
    (unless (file-exists-p repo)
      (make-directory repo t)
      (when (< emacs-major-version 28) (require 'subr-x))
      (condition-case-unless-debug err
    (if-let ((buffer (pop-to-buffer-same-window "*elpaca-bootstrap*"))
	     ((zerop (call-process "git" nil buffer t "clone"
				   (plist-get order :repo) repo)))
	     ((zerop (call-process "git" nil buffer t "checkout"
				   (or (plist-get order :ref) "--"))))
	     (emacs (concat invocation-directory invocation-name))
	     ((zerop (call-process emacs nil buffer nil "-Q" "-L" "." "--batch"
				   "--eval" "(byte-recompile-directory \".\" 0 'force)")))
	     ((require 'elpaca))
	     ((elpaca-generate-autoloads "elpaca" repo)))
	(kill-buffer buffer)
      (error "%s" (with-current-buffer buffer (buffer-string))))
  ((error) (warn "%s" err) (delete-directory repo 'recursive))))
    (unless (require 'elpaca-autoloads nil t)
      (require 'elpaca)
      (elpaca-generate-autoloads "elpaca" repo)
      (load "./elpaca-autoloads")))
  (add-hook 'after-init-hook #'elpaca-process-queues)
  (elpaca `(,@elpaca-order))

  ;; Install use-package support
  (elpaca elpaca-use-package
    ;; Enable :elpaca use-package keyword.
    (elpaca-use-package-mode)
    ;; Assume :elpaca t unless otherwise specified.
    (setq elpaca-use-package-by-default t))

  ;; Block until current queue processed.
  (elpaca-wait)

#+end_src

** Evil Mode

#+begin_src emacs-lisp


       ;; Expands to: (elpaca evil (use-package evil :demand t))
       (use-package evil
           :init      ;; tweak evil's configuration before loading it
           (setq evil-want-integration t) ;; This is optional since it's already set to t by default.
           (setq evil-want-keybinding nil)
           (setq evil-want-C-u-scroll t)
           (setq evil-vsplit-window-right t)
           (setq evil-split-window-below t)
           (evil-mode))
         (use-package evil-collection
           :after evil
           :config
           (setq evil-collection-mode-list '(dashboard dired ibuffer))
           (evil-collection-init))
         (use-package evil-tutor)

       ;;Turns off elpaca-use-package-mode current declartion
       ;;Note this will cause the declaration to be interpreted immediately (not deferred).
       ;;Useful for configuring built-in emacs features.
       ;;(use-package emacs :elpaca nil :config (setq ring-bell-function #'ignore))

       ;; Don't install anything. Defer execution of BODf
       ;;(elpaca nil (message "deferred"))

#+end_src

** Keybindings (General.el)
see "~/.config/emacs/keymaps.org"
#+begin_src emacs-lisp

    (use-package general
        :config
        (general-evil-setup)

    ;; set up 'SPC' as the global leader key
    (general-create-definer ns/leader-keys
    :states '(normal insert visual emacs)
    :keymaps 'override
    :prefix "SPC" ;; set leader
    :global-prefix "C-SPC") ;; access leader in insert mode


    ;; built in emacs functions
    (ns/leader-keys
     "SPC" '(counsel-M-x :wk "Mx"))

    ;; buffer bindings
    (ns/leader-keys
    "b" '(:ignore t :wk "buffer")
    "bb" '(switch-to-buffer :wk "Switch buffer")
    "bi" '(ibuffer :wk "iBuffer")
    "bk" '(kill-this-buffer :wk "Kill this buffer")
    "bn" '(next-buffer :wk "Next buffer")
    "bp" '(previous-buffer :wk "Previous buffer")
    "br" '(revert-buffer :wk "Reload buffer"))

   ;; File Bindings
    (ns/leader-keys
    "f" '(:ignore t :wk "file")
    "f." '(find-file :wk "find file")
    "fc" '(lambda() (interactive) (find-file "~/.config/emacs/config.org") :wk "Open Emacs Config")
    "frr" '((lambda () (interactive)
                  (load-file "~/.config/emacs/init.el")
                  (ignore (elpaca-process-queues)))
              :wk "Reload emacs config"))


    ;; Help Bindings

    (ns/leader-keys
    "h" '(:ignore t :wk "help")
    "hf" '(describe-function :wk "function")
    "hv" '(describe-variable :wk "variable"))

    ;; Org Mode Bindings

    (ns/leader-keys
    "o"  '(:ignore t :wk "org")
    "oc" '(org-capture :wk "org capture")
    "oa" '(org-agenda :wk "org agenda")
    "oo" '(lambda() (interactive) (org-roam-node-find) :wk "find org file")
    "oh" '(lambda() (interactive) (find-file "~/org/Obs_Org/Home.org") :wk "Open Org Index")
    "ow" '(lambda() (interactive) (find-file "~/org/Obs_Org/Work.org") :wk "Open Work Index")

    "oi" '(:ignore t :wk "org insert")
    "oil" '(org-insert-link :wk "org insert link")
    "oir" '(org-roam-node-insert :wk "org insert roam node")
    "oit" '(org-time-stamp-inactive :wk "org insert timestamp (inactive)")

    "ot" '(:ignore t :wk "org toggle")
    "ott" '(org-todo :wk "org toggle todo") ;; replace this with C-t
    "otl" '(org-toggle-link-display :wk "toggle link display(org)"))


    ;; Toggles
    (ns/leader-keys
    "t"  '(:ignore t :wk "toggle")
    "tl" '(display-line-numbers-mode :wk "toggle line numbers")
    "tv" '(vterm-toggle :wk "toggle vterm"))

    ;; Window Bindings
    (ns/leader-keys
    "w" '(:ignore t :wk "Windows")

    ;; Window splits
    "w c" '(evil-window-delete :wk "Close window")
    "w n" '(evil-window-new :wk "New window")
    "w s" '(evil-window-split :wk "Horizontal split window")
    "w v" '(evil-window-vsplit :wk "Vertical split window")

    ;; Window motions
    "w h" '(evil-window-left :wk "Window left")
    "w j" '(evil-window-down :wk "Window down")
    "w k" '(evil-window-up :wk "Window up")
    "w l" '(evil-window-right :wk "Window right")
    "w w" '(evil-window-next :wk "Goto next window"))

    ;; eval actionss
    (ns/leader-keys
    "e" '(:ignore t :wk "Eshell/Evaluate")    
    "e b" '(eval-buffer :wk "Evaluate buffer")
    "e d" '(eval-defun :wk "Evaluate after point")
    "e e" '(eval-expression :wk "Evalu te expression")
    "e l" '(eval-last-sexp :wk "Evaluate before point")
    "e r" '(eval-region :wk "Evaluate region"))

      ;; Avy for vim-sneak like actions
      (ns/leader-keys
        "s" '(evil-avy-goto-char-2 :wk "Search"))
  )


#+end_src
** Startup Settings
Follow symlinks by default
#+begin_src emacs-lisp
  (setq vc-follow-symlinks t)
#+end_src
Diminish most minor modes
#+begin_src emacs-lisp
  (use-package diminish)
#+end_src

* Git
** Magit

#+begin_src emacs-lisp
  (use-package seq)
  (use-package transient
    :after seq)
  (use-package magit
  :after transient)
#+end_src
* Ivy

#+begin_src emacs-lisp

  (use-package counsel
    :after ivy
    :diminish
    :config (counsel-mode))

  (use-package ivy
    :diminish
    :bind
    ;; ivy-resume resumes the last Ivy-based completion.
    (("C-c C-r" . ivy-resume)
     ("C-x B" . ivy-switch-buffer-other-window))
    :custom
    (setq ivy-use-virtual-buffers t)
    (setq ivy-count-format "(%d/%d) ")
    (setq enable-recursive-minibuffers t)
    :config
    (ivy-mode))

  (use-package all-the-icons-ivy-rich
    :ensure t
    :init (all-the-icons-ivy-rich-mode 1))

  (use-package ivy-rich
    :after ivy
    :ensure t
    :init (ivy-rich-mode 1) ;; this gets us descriptions in M-x.
    :custom
    (ivy-virtual-abbreviate 'full
     ivy-rich-switch-buffer-align-virtual-buffer t
     ivy-rich-path-style 'abbrev)
    :config
    (ivy-set-display-transformer 'ivy-switch-buffer
				 'ivy-rich-switch-buffer-transformer))
#+end_src

* Language Support (LSP et. al)

** Company
A generic completion framework

#+begin_src emacs-lisp
  (use-package company
    :defer 2
    :diminish
    :custom
    (company-begin-commands '(self-insert-command))
    (company-idle-delay .1)
    (company-minimum-prefix-length 2)
    (company-show-numbers t)
    (company-tooltip-align-annotations 't)
    (global-company-mode t)
  )

  (use-package company-box
    :after company
    :diminish
    :hook (company-mode . company-box-mode))
#+end_src

** LSP
TODO 
* Org Mode
** Org Bullets
#+begin_src emacs-lisp

  (add-hook 'org-mode-hook 'org-indent-mode)
  (use-package org-bullets)
  (add-hook 'org-mode-hook (lambda () (org-bullets-mode 1)))

#+end_src
** Org Capture Templates
TODO
** Org-Modern
** Org Roam
#+begin_src emacs-lisp
    (use-package org-roam
    :ensure t)
    (setq org-roam-completion-everywhere t)
    (setq org-roam-directory "/Users/noah.sapse/org/Obs_Org/")


    (use-package consult-org-roam
     :ensure t
     :after org-roam
     :init
     (require 'consult-org-roam)
     ;; Activate the minor mode
     (consult-org-roam-mode 1)
     :custom
     ;; Use `ripgrep' for searching with `consult-org-roam-search'
     (consult-org-roam-grep-func #'consult-ripgrep)
     ;; Configure a custom narrow key for `consult-buffer'
     (consult-org-roam-buffer-narrow-key ?r)
     ;; Display org-roam buffers right after non-org-roam buffers
     ;; in consult-buffer (and not down at the bottom)
     (consult-org-roam-buffer-after-buffers t)
     :config
     ;; Eventually suppress previewing for certain functions
     (consult-customize
      consult-org-roam-forward-links
      :preview-key "M-.")
     :bind
    )
#+end_src
** Org Table of Contents
#+begin_src emacs-lisp

  (use-package toc-org
      :commands toc-org-enable
      :init (add-hook 'org-mode-hook 'toc-org-enable))

#+end_src
** Org Tempo
*** Source Code Block Tag Expansion
Org-tempo is not a separate package but a module within org that can be enabled.  Org-tempo allows for '<s' followed by TAB to expand to a begin_src tag.  Other expansions available include:

| Typing the below + TAB | Expands to ...                          |
|------------------------+-----------------------------------------|
| <a                     | '#+BEGIN_EXPORT ascii' … '#+END_EXPORT  |
| <c                     | '#+BEGIN_CENTER' … '#+END_CENTER'       |
| <C                     | '#+BEGIN_COMMENT' … '#+END_COMMENT'     |
| <e                     | '#+BEGIN_EXAMPLE' … '#+END_EXAMPLE'     |
| <E                     | '#+BEGIN_EXPORT' … '#+END_EXPORT'       |
| <h                     | '#+BEGIN_EXPORT html' … '#+END_EXPORT'  |
| <l                     | '#+BEGIN_EXPORT latex' … '#+END_EXPORT' |
| <q                     | '#+BEGIN_QUOTE' … '#+END_QUOTE'         |
| <s                     | '#+BEGIN_SRC' … '#+END_SRC'             |
| <v                     | '#+BEGIN_VERSE' … '#+END_VERSE'         |

#+begin_src emacs-lisp
    (require 'org-tempo)
#+end_src


* Projectile
#+begin_src emacs-lisp

    (use-package projectile
      :config
      (projectile-mode 1)
      :diminish)

#+end_src

* Shells and Terminals
** Eshell
#+begin_src emacs-lisp

  (use-package eshell-syntax-highlighting
    :after esh-mode
    :config
    (eshell-syntax-highlighting-global-mode +1))

  ;; eshell-syntax-highlighting -- adds fish/zsh-like syntax highlighting.
  ;; eshell-rc-script -- your profile for eshell; like a bashrc for eshell.
  ;; eshell-aliases-file -- sets an aliases file for the eshell.
  
  (setq eshell-rc-script (concat user-emacs-directory "eshell/profile")
        eshell-aliases-file (concat user-emacs-directory "eshell/aliases")
        eshell-history-size 5000
        eshell-buffer-maximum-lines 5000
        eshell-hist-ignoredups t
        eshell-scroll-to-bottom-on-input t
        eshell-destroy-buffer-when-process-dies t
        eshell-visual-commands'("bash" "fish" "htop" "ssh" "top" "zsh"))

#+end_src
** Vterm
#+begin_src emacs-lisp

  ;; vterm
  (use-package vterm
    :config
    (setq shell-file-name "/bin/zsh"
            vterm-max-scrollback 5000))

  (setq vterm-timer-delay 0.01)
  ;; vterm-toggle
  (use-package vterm-toggle
    :after vterm
    :config
    (setq vterm-toggle-fullscreen-p nil)
    (setq vterm-toggle-scope 'project)
    (add-to-list 'display-buffer-alist
                 '((lambda (buffer-or-name _)
                       (let ((buffer (get-buffer buffer-or-name)))
                         (with-current-buffer buffer
                           (or (equal major-mode 'vterm-mode)
                               (string-prefix-p vterm-buffer-name (buffer-name buffer))))))
                    (display-buffer-reuse-window display-buffer-at-bottom)
                    ;;(display-buffer-reuse-window display-buffer-in-direction)
                    ;;display-buffer-in-direction/direction/dedicated is added in emacs27
                    ;;(direction . bottom)
                    ;;(dedicated . t) ;dedicated is supported in emacs27
                    (reusable-frames . visible)
                    (window-height . 0.3))))

#+end_src
* Sudo Edit
#+begin_src emacs-lisp
   (use-package sudo-edit
    :config
        (ns/leader-keys
        "fu" '(sudo-edit-find-file :wk "Sudo find file")
        "fU" '(sudo-edit :wk "Sudo edit file")))
#+end_src

* UI Tweaks
** Fonts
#+begin_src emacs-lisp
  (set-face-attribute 'default nil
    :font "JetBrains Mono"
    :height 240
    :weight 'medium)
  (set-face-attribute 'variable-pitch nil
    :font "Ubuntu"
    :height 250
    :weight 'medium)
  (set-face-attribute 'fixed-pitch nil
    :font "JetBrains Mono"
    :height 240
    :weight 'medium)
  ;; Makes commented text and keywords italics.
  ;; This is working in emacsclient but not emacs.
  ;; Your font must have an italic face available.
  (set-face-attribute 'font-lock-comment-face nil
    :slant 'italic)
  (set-face-attribute 'font-lock-keyword-face nil
    :slant 'italic)

  ;; This sets the default font on all graphical frames created after restarting Emacs.
  ;; Does the same thing as 'set-face-attribute default' above, but emacsclient fonts
  ;; are not right unless I also add this method of setting the default font.
  (add-to-list 'default-frame-alist '(font . "JetBrains Mono-24"))

  ;; per internet helps with GUI font
#+end_src

** Icons
#+begin_src emacs-lisp

  (use-package all-the-icons
    :ensure t
    :if (display-graphic-p))

  (use-package all-the-icons-dired
    :hook (dired-mode . (lambda () (all-the-icons-dired-mode t))))

#+end_src

** Line Numbers
enable line numbers globally but disable them in certain modes
#+begin_src emacs-lisp

  (global-display-line-numbers-mode 1)
  (global-visual-line-mode t)

  (dolist (mode '(org-mode-hook
                      term-mode-hook
                      vterm-mode-hook
                      shell-mode-hook
                      eshell-mode-hook))
        (add-hook mode (lambda() (display-line-numbers-mode 0))))

#+end_src
** Modeline (Doom Modeline)
#+begin_src emacs-lisp

  (use-package doom-modeline
    :ensure t
    :config
    (setq doom-modeline-height 40)
    :init (doom-modeline-mode 1))

#+end_src

** Themes 
#+begin_src emacs-lisp

      (use-package doom-themes
        :ensure t
        :config
        ;; Global settings (defaults)
        (setq doom-themes-enable-bold t    ; if nil, bold is universally disabled
              doom-themes-enable-italic t) ; if nil, italics is universally disabled
        (load-theme 'doom-one t)

        ;; Corrects (and improves) org-mode's native fontification.
        (doom-themes-org-config))

       (use-package nano-theme
        :ensure nil
        :defer t)
#+end_src

* Which Key
#+begin_src emacs-lisp
      (use-package which-key
	:init
	  (which-key-mode 1)
	:config
	(setq which-key-side-window-location 'bottom
		which-key-sort-order #'which-key-key-order-alpha
		which-key-sort-uppercase-first nil
		which-key-add-column-padding 1
		which-key-max-display-columns nil
		which-key-min-display-lines 6
		which-key-side-window-slot -10
		which-key-side-window-max-height 0.25
		which-key-idle-delay 0.8
		which-key-max-description-length 25
		which-key-allow-imprecise-window-fit t
		which-key-separator " → " )
	 :diminish
       )

#+end_src


emacs/.config/emacs/.#config.org
noah.sapse@AMB-FP04FY.30093

emacs/.config/emacs/config.el
(defvar elpaca-installer-version 0.7)
(defvar elpaca-directory (expand-file-name "elpaca/" user-emacs-directory))
(defvar elpaca-builds-directory (expand-file-name "builds/" elpaca-directory))
(defvar elpaca-repos-directory (expand-file-name "repos/" elpaca-directory))
(defvar elpaca-order '(elpaca :repo "https://github.com/progfolio/elpaca.git"
                              :ref nil :depth 1
                              :files (:defaults "elpaca-test.el" (:exclude "extensions"))
                              :build (:not elpaca--activate-package)))
(let* ((repo  (expand-file-name "elpaca/" elpaca-repos-directory))
       (build (expand-file-name "elpaca/" elpaca-builds-directory))
       (order (cdr elpaca-order))
       (default-directory repo))
  (add-to-list 'load-path (if (file-exists-p build) build repo))
  (unless (file-exists-p repo)
    (make-directory repo t)
    (when (< emacs-major-version 28) (require 'subr-x))
    (condition-case-unless-debug err
        (if-let ((buffer (pop-to-buffer-same-window "*elpaca-bootstrap*"))
                 ((zerop (apply #'call-process `("git" nil ,buffer t "clone"
                                                 ,@(when-let ((depth (plist-get order :depth)))
                                                     (list (format "--depth=%d" depth) "--no-single-branch"))
                                                 ,(plist-get order :repo) ,repo))))
                 ((zerop (call-process "git" nil buffer t "checkout"
                                       (or (plist-get order :ref) "--"))))
                 (emacs (concat invocation-directory invocation-name))
                 ((zerop (call-process emacs nil buffer nil "-Q" "-L" "." "--batch"
                                       "--eval" "(byte-recompile-directory \".\" 0 'force)")))
                 ((require 'elpaca))
                 ((elpaca-generate-autoloads "elpaca" repo)))
            (progn (message "%s" (buffer-string)) (kill-buffer buffer))
          (error "%s" (with-current-buffer buffer (buffer-string))))
      ((error) (warn "%s" err) (delete-directory repo 'recursive))))
  (unless (require 'elpaca-autoloads nil t)
    (require 'elpaca)
    (elpaca-generate-autoloads "elpaca" repo)
    (load "./elpaca-autoloads")))
(add-hook 'after-init-hook #'elpaca-process-queues)
(elpaca `(,@elpaca-order))

;; Install use-package support
(elpaca elpaca-use-package
  ;; Enable :elpaca use-package keyword.
  (elpaca-use-package-mode)
  ;; Assume :elpaca t unless otherwise specified.
  (setq elpaca-use-package-by-default t))

;; Block until current queue processed.
(elpaca-wait)

;;When installing a package which modifies a form used at the top-level
;;(e.g. a package which adds a use-package key word),
;;use `elpaca-wait' to block until that package has been installed/configured.
;;For example:
;;(use-package general :demand t)
;;(elpaca-wait)

;; Expands to: (elpaca evil (use-package evil :demand t))
(use-package evil
    :init      ;; tweak evil's configuration before loading it
    (setq evil-want-integration t) ;; This is optional since it's already set to t by default.
    (setq evil-want-keybinding nil)
    (setq evil-want-C-u-scroll t)
    (setq evil-vsplit-window-right t)
    (setq evil-split-window-below t)
    (evil-mode))
  (use-package evil-collection
    :after evil
    :config
    (setq evil-collection-mode-list '(dashboard dired ibuffer))
    (evil-collection-init))
  (use-package evil-tutor)

;;Turns off elpaca-use-package-mode current declartion
;;Note this will cause the declaration to be interpreted immediately (not deferred).
;;Useful for configuring built-in emacs features.
;;(use-package emacs :elpaca nil :config (setq ring-bell-function #'ignore))

;; Don't install anything. Defer execution of BODY
;;(elpaca nil (message "deferred"))

(use-package general
  :config
  (general-evil-setup)

  ;; set up 'SPC' as the global leader key
  (general-create-definer ns/leader-keys
    :states '(normal insert visual emacs)
    :keymaps 'override
    :prefix "SPC" ;; set leader
    :global-prefix "M-SPC") ;; access leader in insert mode

  ;; buffer actions
  (ns/leader-keys
    "b" '(:ignore t :wk "buffer")
    "bb" '(switch-to-buffer :wk "Switch buffer")
    "bi" '(ibuffer :wk "iBuffer")
    "bk" '(kill-this-buffer :wk "Kill this buffer")
    "bn" '(next-buffer :wk "Next buffer")
    "bp" '(previous-buffer :wk "Previous buffer")
    "br" '(revert-buffer :wk "Reload buffer"))

  ;; file actions
  (ns/leader-keys
    "." '(find-file :wk "find file")
    "fc" '(lambda() (interactive) (find-file "~/.config/emacs/config.org") :wk "Open Emacs Config")
    )

  ;; lisp evaluation
  ;;(ns/leader-keys
  ;; "e" ("ignore t :wk "evaluate"))
)

(use-package which-key
  :init
    (which-key-mode 1)
  :config
  (setq which-key-side-window-location 'bottom
          which-key-sort-order #'which-key-key-order-alpha
          which-key-sort-uppercase-first nil
          which-key-add-column-padding 1
          which-key-max-display-columns nil
          which-key-min-display-lines 6
          which-key-side-window-slot -10
          which-key-side-window-max-height 0.25
          which-key-idle-delay 0.8
          which-key-max-description-length 25
          which-key-allow-imprecise-window-fit t
          which-key-separator " → " ))

(use-package toc-org
    :commands toc-org-enable
    :init (add-hook 'org-mode-hook 'toc-org-enable))

(add-hook 'org-mode-hook 'org-indent-mode)
(use-package org-bullets)
(add-hook 'org-mode-hook (lambda () (org-bullets-mode 1)))
